streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -440,6 +440,18 @@ public void commit() {
         commit(true);
     }
 
+    private long getNextPartitionTime(final TopicPartition partition) {
+        final OffsetAndMetadata oldCommittedMetadata = consumer.committed(partition);
+        final long metadataTimestamp;
+        if (oldCommittedMetadata != null && oldCommittedMetadata.metadata().length() != 0) {
```

Comment:
is old metadata missing expected after we start off? Might be useful to add a debug log or trace if this is not normal.

Reply:
Well, since before, we never stored any metadata in Kafka log, especially relating to committed timestamps. There is a possibility that an old version of OffsetAndMetadata is committed where it doesn't contain the committed timestamp, so I suppose this is expected behavior.